File: gunhoflash/rotate-image.py
from typing import List

class Solution:
  def rotate(self, matrix: List[List[int]]) -> None:
    n = len(matrix)
    for i in range((n + 1) // 2):
      for j in range(i, n - i - 1):
        [
          matrix[i][j],
          matrix[n - j - 1][i],
          matrix[n - i - 1][n - j - 1],
          matrix[j][n - i - 1]
        ] = [
          matrix[n - j - 1][i],
          matrix[n - i - 1][n - j - 1],
          matrix[j][n - i - 1],
          matrix[i][j]
        ]

File: gunhoflash/rotate-image.ts
function rotate(matrix: number[][]): void {
  const n = matrix.length;
  for (let i = 0; i < n / 2; i++) {
    for (let j = i; j < n - i - 1; j++) {
      const temp = matrix[i][j];
      matrix[i][j] = matrix[n - j - 1][i];
      matrix[n - j - 1][i] = matrix[n - i - 1][n - j - 1];
      matrix[n - i - 1][n - j - 1] = matrix[j][n - i - 1];
      matrix[j][n - i - 1] = temp;
    }
  }
}
